include/world_builder/features/mantle_layer_models/grains/random_uniform_distribution_deflected.h

```diff
@@ -0,0 +1,115 @@
+#define WORLD_BUILDER_FEATURES_MANTLE_LAYER_MODELS_GRAINS_RANDOM_UNIFORM_DISTRIBUTION_DEFLECTED_H
+
+
+#include "world_builder/features/mantle_layer_models/grains/interface.h"
+#include "world_builder/objects/surface.h"
+
+namespace WorldBuilder
+{
+
+  namespace Features
+  {
+    namespace MantleLayerModels
+    {
+      namespace Grains
+      {
+        /**
+         * This class represents a mantle layer and can implement
+         * submodules for temperature and grains. These submodules determine
+         * what the returned temperature or grains of the temperature and grains
+         * functions of this class will be.
+         */
+        class RandomUniformDistributionDeflected final: public Interface
+        {
+          public:
+            /**
+             * constructor
+             */
+            RandomUniformDistributionDeflected(WorldBuilder::World *world);
+
+            /**
+             * Destructor
+             */
+            ~RandomUniformDistributionDeflected() override final;
+
+            /**
+             * declare and read in the world builder file into the parameters
+             * class
+             */
+            static void declare_entries(Parameters &prm,
+                                        const std::string &parent_name = "");
+
+            /**
+             * declare and read in the world builder file into the parameters
+             * class
+             */
+            static void declare_grain_size_model_entries(
+              Parameters &prm, const std::string &parent_name,
+              const std::vector<std::string> &required_entries);
+
+            /**
+             * declare and read in the world builder file into the parameters
+             * class
+             */
+            static void
+            declare_fixed_size_model_entries(Parameters &prm,
+                                             const std::string &parent_name = "");
+
+            /**
+             * declare and read in the world builder file into the parameters
+             * class
+             */
+            void parse_entries(Parameters &prm, const std::vector<Point<2>> &coordinates) override final;
+
+            /**
+             * Returns a grains based on the given position, composition (e.g.
+             * olivine and/or enstatite)depth in the model, gravity and current grains.
+             */
+            WorldBuilder::grains
+            get_grains(const Point<3> &position,
+                       const Objects::NaturalCoordinate &position_in_natural_coordinates,
+                       const double depth,
+                       const unsigned int composition_number,
+                       WorldBuilder::grains grains,
+                       const double feature_min_depth,
+                       const double feature_max_depth) const override final;
+
+          private:
+            // random uniform distribution deflected grains submodule parameters
```

Comment:
I would delete this comment as it doesn't add any useful information.

Reply:
Removed.